github-actions/DotNet.GitHubAction/DotNet.CodeAnalysis/FileUtilities.cs

```diff
@@ -0,0 +1,401 @@
+    internal static class FileUtilities
+    {
+        /// <summary>
+        /// Resolves relative path and returns absolute path.
+        /// The method depends only on values of its parameters and their implementation (for fileExists).
+        /// It doesn't itself depend on the state of the current process (namely on the current drive directories) or 
+        /// the state of file system.
+        /// </summary>
+        /// <param name="path">
+        /// Path to resolve.
+        /// </param>
+        /// <param name="basePath">
+        /// Base file path to resolve CWD-relative paths against. Null if not available.
+        /// </param>
+        /// <param name="baseDirectory">
+        /// Base directory to resolve CWD-relative paths against if <paramref name="basePath"/> isn't specified. 
+        /// Must be absolute path.
+        /// Null if not available.
+        /// </param>
+        /// <param name="searchPaths">
+        /// Sequence of paths used to search for unqualified relative paths.
+        /// </param>
+        /// <param name="fileExists">
+        /// Method that tests existence of a file.
+        /// </param>
+        /// <returns>
+        /// The resolved path or null if the path can't be resolved or does not exist.
+        /// </returns>
+        internal static string? ResolveRelativePath(
+            string path,
+            string? basePath,
+            string? baseDirectory,
+            IEnumerable<string> searchPaths,
+            Func<string, bool> fileExists)
+        {
+            string? combinedPath;
+            var kind = PathUtilities.GetPathKind(path);
+            if (kind == PathKind.Relative)
+            {
+                // first, look in the base directory:
+                baseDirectory = GetBaseDirectory(basePath, baseDirectory);
+                if (baseDirectory != null)
+                {
+                    combinedPath = PathUtilities.CombinePathsUnchecked(baseDirectory, path);
+                    if (fileExists(combinedPath))
+                    {
+                        return combinedPath;
+                    }
+                }
+
+                // try search paths:
+                foreach (var searchPath in searchPaths)
+                {
+                    combinedPath = PathUtilities.CombinePathsUnchecked(searchPath, path);
+                    if (fileExists(combinedPath))
+                    {
+                        return combinedPath;
+                    }
+                }
+
+                return null;
+            }
+
+            combinedPath = ResolveRelativePath(kind, path, basePath, baseDirectory);
+            if (combinedPath != null)
+            {
+                if (fileExists(combinedPath))
+                {
+                    return combinedPath;
+                }
+            }
+
+            return null;
+        }
+
+        internal static string? ResolveRelativePath(string? path, string? baseDirectory)
```

Comment:
I'm a fan of expression bodied members here, but that's a style issue and a nit.